src/bin/blob_fs_clean_up.js

```diff
@@ -0,0 +1,355 @@
+                    if (typeof object.atr[key] === 'string' &&
+                        REGEXP.BLOB_HASH.test(object.atr[key])) {
+                        metaHashes[object.atr[key]] = true;
+                    }
+                }
+            }
+            next();
+        },
+        i, j,
+        allProjects;
+
+    metadatastorage.getProjects()
+        .then(function (all) {
+            var promises = [];
+
+            allProjects = all;
+            for (i = 0; i < allProjects.length; i += 1) {
+                promises.push(storage.traverse({
+                    projectId: allProjects[i]._id,
+                    visitFn: checkIndividualRecord
+                }));
+            }
+            return Q.all(promises);
+        })
+        .then(function () {
+            var promises = [];
+
+            for (i in metaHashes) {
+                promises.push(gatherSoftLinks(blobClient, i));
+            }
+            return Q.all(promises);
+        })
+        .then(function (results) {
+            results = results || [];
+            for (i = 0; i < results.length; i += 1) {
+                for (j = 0; j < results[i].length; j += 1) {
+                    metaHashes[results[i][j]] = true;
+                }
+            }
+            deferred.resolve(metaHashes);
+        })
+        .catch(deferred.reject);
+
+    return deferred.promise;
+}
+
+function getUnusedMetaHashes(blobClient, metadatastorage, storage) {
+    var deferred = Q.defer(),
+        allHashesArray,
+        usedHashesObject,
+        i;
+
+    blobClient.listObjects('wg-metadata')
+        .then(function (all) {
+            allHashesArray = all;
+            return getUsedMetaHashes(metadatastorage, storage, blobClient);
+        })
+        .then(function (used) {
+            usedHashesObject = used;
+            i = allHashesArray.length;
+            while (i--) {
+                if (usedHashesObject[allHashesArray[i]] === true) {
+                    allHashesArray.splice(i, 1);
+                }
+            }
+            deferred.resolve(allHashesArray);
+        })
+        .catch(deferred.reject);
+
+    return deferred.promise;
+}
+
+function gatherSoftLinks(blobClient, metaHash) {
+    var deferred = Q.defer(),
+        softLinks = [],
+        complexEntry;
+
+    blobClient.getMetadata(metaHash)
+        .then(function (metaData) {
+            if (metaData.contentType === CONTENT_TYPES.SOFT_LINK) {
+                softLinks.push(metaData.content);
+            } else if (metaData.contentType === CONTENT_TYPES.COMPLEX) {
+                for (complexEntry in metaData.content) {
+                    if (metaData.content[complexEntry].contentType === CONTENT_TYPES.SOFT_LINK) {
+                        softLinks.push(metaData.content[complexEntry].content);
+                    }
+                }
+            }
+            deferred.resolve(softLinks);
+        })
+        .catch(deferred.reject);
```

Comment:
You could expect to see some errors here. The only check for the assets is if it is a blob-hash.